integrations/wagmi-v2/src/index.ts

```diff
@@ -63,3 +84,131 @@ export function injectedWithSapphire(
 		},
 	} as InjectedParameters);
 }
+
+/**
+ * Creates a wagmi configuration with wrapped EIP-6963 Sapphire encryption layer.
+ *
+ * In case you are only using Sapphire chain:
+ *
+ * ```typescript
+ * import { createSapphireConfig } from '@oasisprotocol/sapphire-wagmi-v2';
+ *
+ * export const config = createSapphireConfig({
+ *   sapphireConfig: {
+ *     replaceProviders: true,
+ *   }
+ *   ...
+ * });
+ * ```
+ *
+ * In case your dApp is multichain, the below example will create duplicate connectors for Sapphire:
+ *
+ * ```typescript
+ * import { createSapphireConfig } from '@oasisprotocol/sapphire-wagmi-v2';
+ *
+ * export const config = createSapphireConfig({
+ *   sapphireConfig: {
+ *     replaceProviders: false,
+ *     // Define which providers you want to wrap via RDNS
+ *     wrappedProvidersFilter: (rdns) => ['io.metamask'].includes(rdns)
+ *   }
+ *   ...
+ * });
+ * ```
+ *
+ * @param {CreateConfigParameters<chains, transports, connectorFns> & CreateSapphireConfigParameters} parameters - Extended wagmi parameters,
+ * with sapphireConfig
+ * @return {Config<chains, transports, connectorFns>} Wagmi config
+ */
+export function createSapphireConfig<
+	chains extends readonly [Chain, ...Chain[]],
+	transports extends Record<chains[number]["id"], Transport>,
+	connectorFns extends readonly CreateConnectorFn[],
+>(
+	parameters: CreateConfigParameters<chains, transports, connectorFns> &
+		CreateSapphireConfigParameters,
+): Config<chains, transports, connectorFns> {
+	const { sapphireConfig, ...restParameters } = parameters;
+	const {
+		replaceProviders = false,
+		wrappedProvidersFilter = () => true,
+		wrap: wrapOptions,
+	} = sapphireConfig;
+
+	const _addEventListener = EventTarget.prototype.addEventListener;
+	Object.defineProperty(EventTarget.prototype, "addEventListener", {
+		value: function (
+			type: string,
+			callback: EventListenerOrEventListenerObject | null,
+			options?: AddEventListenerOptions | boolean,
+		): void {
+			if (type === EIP6963_ANNOUNCE_PROVIDER_EVENT_NAME) {
+				_addEventListener.call(
+					this,
+					type,
+					(event: Event) => {
+						let patchCustomEvent = null;
```

Comment:
This looks like it breaks removeEventListener (callback function won't match inline patch function)

Reply:
If removing EIP6963_ANNOUNCE_PROVIDER_EVENT_NAME listeners isn't common then I would just add a headsup comment

Reply:
I haven't seen a codebase, that would try to remove the event. But yeah, I see the potential issue. Thats why I didn't add "restore" to original `_addEventListener`, which could cause memory leaks, because I did not expect that someone would try to remove the event listener. 

Either way, I added the comment here: https://github.com/oasisprotocol/sapphire-paratime/pull/565. I hope that is enough for now, and if there is an use-case for it. I am happy to implement more robust solution, that would "reverse" event patching.